src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -1054,17 +1066,20 @@ public final class PlaceholderExecutable implements ContinuableExecutable, Acces
                             LOGGER.fine(() -> "interrupted " + cookie + " in " + runId);
                             Timer.get().submit(() -> { // JENKINS-46738
                                 Executor thisExecutor = /* AsynchronousExecution. */ getExecutor();
+                                AtomicReference<Boolean> cancelledBodyExecution = new AtomicReference(false);
                                 withExecution(execution -> {
                                     BodyExecution body = execution.body;
                                     if (body != null) {
                                         body.cancel(thisExecutor != null ? thisExecutor.getCausesOfInterruption().toArray(new CauseOfInterruption[0]) : new CauseOfInterruption[0]);
-                                    } else { // anomalous state; perhaps build already aborted but this was left behind; let user manually cancel executor slot
```

Comment:
If the execution does not exist, because the build is complete or corrupted or whatever, we never even enter this block and so this logic wasn't reachable.